Userland/Libraries/LibGL/Shaders/Shader.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/NonnullRefPtr.h>
+#include <AK/Optional.h>
+#include <AK/OwnPtr.h>
+#include <AK/RefCounted.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+#include <LibGL/GL/gl.h>
+#include <LibGLSL/ObjectFile.h>
+
+namespace GL {
+
+class Shader final : public RefCounted<Shader> {
+public:
+    static NonnullRefPtr<Shader> create(GLenum shader_type);
+
+    void clear_sources() { m_sources.clear(); }
+    void add_source(StringView source_code) { m_sources.append(source_code); }
```

Comment:
This code needs some help regarding the String --> DeprecatedString refactor. At least m_sources and m_info_log here should be converted to DeprecatedString entries until we can refactor later.